bin/test_integration

```diff
@@ -0,0 +1,135 @@
+#!/bin/bash
+
+set -eo pipefail
+
+# Run local integration tests (i.e. tests that don't require
+# integration with PCF/Tanzu).
+
+bin_dir="$(dirname $0)"
+temp_dir="$bin_dir/../tmp"
+. "$bin_dir"/utils.sh
+
+function cleanup {
+  announce 'Removing test environment'
+  docker-compose down --rmi 'local' --volumes
+  rm -f "$temp_dir"/pids/server*.pid
+}
+trap cleanup EXIT
+
+SERVICE_BROKERS='service-broker-bad-url service-broker-bad-key service-broker-follower-url'
+
+function main() {
+  cleanup
+  startConjur
+
+  # Run Conjur V5 Tests
+  loadPolicy5
+  startServiceBrokers5
+  runTests
+  cleanUpServiceBrokers
+
+  # Run Conjur V4 Tests
+  loadPolicy4
+  startServiceBrokers4
+  runTests
+  cleanUpServiceBrokers
+}
+
+function startConjur() {
+  announce "Starting Conjur environment"
+  docker-compose up -d pg conjur_4 conjur_5
+}
+
+function loadPolicy5() {
+  announce "Waiting for Conjur v5 to come up, and loading policy..."
+
+  docker-compose exec -T conjur_5 conjurctl wait -r 30 -p 80
+
+  api_key=$(docker-compose exec -T conjur_5 bash -c 'rails r "puts Role[%Q{cucumber:user:admin}].api_key" 2>/dev/null')
+  export CONJUR_AUTHN_API_KEY="$api_key"
+
+  # load the pcf policy for the non-empty CONJUR_POLICY test
+  docker-compose run --rm --entrypoint bash client -c "conjur policy load root /app/policy/policy.yml"
+  docker-compose run --rm --entrypoint bash client -c "conjur policy load root /app/policy/cf.yml"
+  docker-compose run --rm --entrypoint bash client -c "conjur policy load cf /app/policy/org.yml"
+  docker-compose run --rm --entrypoint bash client -c "conjur policy load cf /app/policy/space.yml"
+  docker-compose run --rm --entrypoint bash client -c "conjur policy load cf /app/policy/app.yml"
+}
+
+function startServiceBrokers5() {
+  export CONJUR_VERSION=5
+  export CONJUR_APPLIANCE_URL=http://conjur_5
+  export CONJUR_FOLLOWER_URL=http://conjur_5-follower
+  export CONJUR_SSL_CERTIFICATE=""
+
+  admin_api_key=$(docker-compose exec -T conjur_5 bash -c 'rails r "puts Role[%Q{cucumber:user:admin}].api_key" 2>/dev/null')
+  export CONJUR_AUTHN_API_KEY="$admin_api_key"
+  docker-compose up -d $SERVICE_BROKERS
+
+  export CONJUR_POLICY=cf
+
+  export CONJUR_AUTHN_API_KEY="$(docker-compose exec -T conjur_5 bash -c 'rails r "puts Role[%Q{cucumber:host:cf-service-broker}].api_key" 2>/dev/null')"
```

Comment:
Query: Does `declare -x` silence the codeclimate issues here about masking return values? Or will we have to move them to separate lines?

Reply:
I think we want to do the declare/assign on separate lines... I can't see why we'd want to suppress an error here and continue with a (yikes!) null CONJUR_AUTHN_API_KEY.